platform/src/main/java/org/hillview/sketches/Centroids.java

```diff
@@ -0,0 +1,101 @@
+package org.hillview.sketches;
+
+import org.hillview.dataset.api.IJson;
+import org.hillview.table.api.IMembershipSet;
+import org.hillview.table.api.ITable;
+import org.hillview.utils.BlasConversions;
+import org.jblas.DoubleMatrix;
+import org.jblas.ranges.AllRange;
+import org.jblas.ranges.PointRange;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * This class represents a set of centroids of a partitioning of a table. The number of centroids should not be too
+ * large, as they are shipped over the network.
+ */
+public class Centroids implements IJson {
+    /**
+     * Encodes the position of the centroids. Every row is a centroid, and the columns are the dimensions.
+     * Each partition has a centroid, which is simply the mean of the nD points in the partition.
+     * Should be computed with computeCentroids() by the head node.
+     */
+    @Nullable
+    public DoubleMatrix centroids;
```

Comment:
Does this serialize as JSON properly?

Reply:
DoubleMatrix implements Serializable, so it should, right? Anyway, it doesn't matter now, since I've refactored it to not use the JBLAS classes. Will push the changes when I finish updating the doc.

Reply:
I see now that this is not immediately evident. We should refrain from sending `DoubleMatrix` objects to the client anyway, so this is fine.